internal/pkg/fleetapi/action.go

```diff
@@ -65,45 +68,59 @@ type ScheduledAction interface {
+	// the retry_attempt number is meant to be an internal counter for the elastic-agent and not communicated to fleet-server or ES.
 	SetRetryAttempt(int)
 	// SetStartTime sets the start_time of the action to the specified value.
 	// this is used by the action-retry mechanism.
 	SetStartTime(t time.Time)
 	// GetError returns the error that is associated with the retry.
 	// If it is a retryable action fleet-server should mark it as such.
-	// Otherwise fleet-server should mark the action as failed.
+	// Otherwise, fleet-server should mark the action as failed.
 	GetError() error
 	// SetError sets the retryable action error
 	SetError(error)
 }
 
 type Signed struct {
-	Data      string `yaml:"data" json:"data" mapstructure:"data"`
-	Signature string `yaml:"signature" json:"signature" mapstructure:"signature"`
-}
-
-// FleetAction represents an action from fleet-server.
-// should copy the action definition in fleet-server/model/schema.json
-type FleetAction struct {
-	ActionID         string          `yaml:"action_id" json:"id"` // NOTE schema defines this as action_id, but fleet-server remaps it to id in the json response to agent check-in.
-	ActionType       string          `yaml:"type,omitempty" json:"type,omitempty"`
-	InputType        string          `yaml:"input_type,omitempty" json:"input_type,omitempty"`
-	ActionExpiration string          `yaml:"expiration,omitempty" json:"expiration,omitempty"`
-	ActionStartTime  string          `yaml:"start_time,omitempty" json:"start_time,omitempty"`
-	Timeout          int64           `yaml:"timeout,omitempty" json:"timeout,omitempty"`
-	Data             json.RawMessage `yaml:"data,omitempty" json:"data,omitempty"`
-	Retry            int             `json:"retry_attempt,omitempty" yaml:"retry_attempt,omitempty"` // used internally for serialization by elastic-agent.
-	//Agents []string // disabled, fleet-server uses this to generate each agent's actions
-	//Timestamp string // disabled, agent does not care when the document was created
-	//UserID string // disabled, agent does not care
-	//MinimumExecutionDuration int64 // disabled, used by fleet-server for scheduling
-	Signed *Signed `yaml:"signed,omitempty" json:"signed,omitempty"`
+	Data      string `json:"data" yaml:"data" mapstructure:"data"`
+	Signature string `json:"signature" yaml:"signature"  mapstructure:"signature"`
+}
+
+// NewAction returns a new, zero-value, action of the type defined by 'actionType'
+// or an ActionUnknown with the 'OriginalType' field set to 'actionType' if the
+// type is not valid.
+func NewAction(actionType string) Action {
+	var action Action
+
+	// keep the case statements alphabetically sorted
+	switch actionType {
+	case ActionTypeCancel:
+		action = &ActionCancel{}
+	case ActionTypeDiagnostics:
+		action = &ActionDiagnostics{}
+	case ActionTypeInputAction:
+		// Only INPUT_ACTION type actions could possibly be signed https://github.com/elastic/elastic-agent/pull/2348
```

Comment:
This comment will eventually be out of date if it isn't already. I think. the intent is to sign all of the actions that can result in endpoint be removed from the policy at minimum.